pkg/network/filter/packet_source_linux.go

```diff
@@ -108,7 +128,7 @@ func (p *AFPacketSource) VisitPackets(exit <-chan struct{}, visit func([]byte, t
 			return err
 		}
 
-		if err := visit(data, stats.Timestamp); err != nil {
+		if err := visit(data, stats.AncillaryData[0].(afpacket.AncillaryPktType).Type, stats.Timestamp); err != nil {
```

Comment:
Adding an enum makes this even more ugly since I have to cast this now that enum type.

Reply:
It's less about cleaning up this line, and more about having the callback take something well typed instead of a uint8.

Reply:
The enum would still require me to map every value from the OS to the enum, or I cast to the enum here and then we get a value that is potentially not represented by the enum.

The need here is to pass OS specific info about the packet; I think a better way to do that would be to define OS-specific (or packet source specific) data structures instead of merely an anonymous `uint8`. See https://github.com/DataDog/datadog-agent/pull/27100/commits/ab537d4014d57a52fab2ad4c223587bf3dd47d3f